doc/src/Developer_write_pair.rst

```diff
@@ -0,0 +1,1196 @@
+      init specific to this pair style
+   ------------------------------------------------------------------------- */
+
+   void PairTersoff::init_style()
+   {
+     if (atom->tag_enable == 0)
+       error->all(FLERR,"Pair style Tersoff requires atom IDs");
+     if (force->newton_pair == 0)
+       error->all(FLERR,"Pair style Tersoff requires newton pair on");
+
+     // need a full neighbor list
+
+     neighbor->add_request(this,NeighConst::REQ_FULL);
+   }
+
+Computing forces from the neighbor list
+"""""""""""""""""""""""""""""""""""""""
+
+Computing forces for a many-body potential is usually more complex than
+for a pair-wise additive potential and there are multiple components.
+For Tersoff, there is a pair-wise additive two-body term (two nested
+loops over indices *i* and *j*) and a three-body term (three nested
+loops over indices *i*, *j*, and *k*).  Since the neighbor list has
+all neighbors up to the maximum cutoff (for the two-body term), but
+the three-body interactions have a significantly shorter cutoff,
+a "short neighbor list" is also constructed at the same time while computing
+the two-body term and looping over the neighbor list for the first time.
+
+.. code-block:: c++
+
+   if (rsq < cutshortsq) {
+     neighshort[numshort++] = j;
+     if (numshort >= maxshort) {
+       maxshort += maxshort/2;
+       memory->grow(neighshort,maxshort,"pair:neighshort");
+     }
+   }
+
+For the two-body term, only a half neighbor list would be needed, even
+though we have requested a full list (for the three-body loops).
+Rather than computing all interactions twice, we skip over half of
+the entries.  This is done in a slightly complex way to make certain
+the same choice is made across all subdomains and so that there is
+no load imbalance introduced.
+
+.. code-block:: c++
+
+   jtag = tag[j];
+   if (itag > jtag) {
+     if ((itag+jtag) % 2 == 0) continue;
+   } else if (itag < jtag) {
+     if ((itag+jtag) % 2 == 1) continue;
+   } else {
+     if (x[j][2] < x[i][2]) continue;
+     if (x[j][2] == ztmp && x[j][1] < ytmp) continue;
+     if (x[j][2] == ztmp && x[j][1] == ytmp && x[j][0] < xtmp) continue;
+   }
+
+For the three-body term, there is one additional nested loop and it uses
+the "short" neighbor list, accumulated previously.
+
+.. code-block:: c++
+
+   // three-body interactions
+   // skip immediately if I-J is not within cutoff
+   double fjxtmp,fjytmp,fjztmp;
+
+   for (jj = 0; jj < numshort; jj++) {
+     j = neighshort[jj];
+     jtype = map[type[j]];
+
+     [...]
+
+     for (kk = 0; kk < numshort; kk++) {
+       if (jj == kk) continue;
+       k = neighshort[kk];
+       ktype = map[type[k]];
+
+       [...]
+     }
+   [...]
+
+
+Reading potential parameters
+""""""""""""""""""""""""""""
+
+For the Tersoff potential, the parameters are listed in a file and associated
+with triples for elements.  Thus, the ``coeff()`` function has to do three
+tasks, each of which is delegated to a function:
```

Comment:
explain the * * requirement of the pair coeff command for many manybody potentials.
Which means that coeff() is only called once, unlike for pairwise styles